Added starters page

geezhawk opened the pull request:
Added easy first commit projects as a submodule within the site.

danlrobertson commented on _layouts/default.html:
```diff
@@ -9,6 +9,14 @@
         <!-- Javascript files -->
         <script src="/js/jquery.min.js"></script>
         <script src="/js/bootstrap.min.js"></script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/react/0.14.7/react.min.js"
```
This is going to add react to pages that possibly do not need it right? Could we add another layout for this? Or are we going to use react more?

geezhawk replied:
Yeah that's something I was wondering about. I'm not too familiar with the way that jekyll builds the pages so I'm not sure if we can add a headers section that appends headers to the default layout, or if we have to create an entirely new layout. For now I'll go in and create a new layout just for this page that includes all the necessary dependencies (ie momentjs and react) that are necessary for the starters page but aren't needed elsewhere.

danlrobertson commented on styles/main.scss:
```diff
@@ -46,3 +46,92 @@
 	margin-top: 2em;
 }
+
+.starters {
```
:+1: for the starters section

geezhawk commented:
I'm wondering if submodules are really the best way to approach the issue of adding a starters page here. Our fork differs enough from `servo/servo-starters` that any upstream changes they made would create a lot of merge conflicts if we tried to add them in. + as this SO answer points out, git submodules are pretty complicated, and it could be difficult to ensure that everyone is looking at the same commit when they pull the project. Given the complexity and the fact that merging upstream changes would be so tricky, I'm inclined to say we just include this code in the `anidata.github.io` repo and do away with the servo-starters fork entirely.

I could be persuaded otherwise though :)

geezhawk commented:
Another reason to just make the starters code part of that `anidata.github.io` repo is that it can't be run independently since it depends on being built out by jekyll. I think if we're wanting to integrate it into the anidata.github.io site instead of having it as a stand-alone page, the submodule approach doesn't make as much sense as just building the code into the `anidata.github.io` repo.

danlrobertson commented:
> Our fork differs enough from servo/servo-starters that any upstream changes they made would create a lot of merge conflicts

Totally understandable. I like submodules for the history. I'm 100% cool with a good old `cp -r`. @gwpjp  thoughts?

geezhawk commented:
Switched away from git submodule pattern and moved react + momentjs dependencies to a new layout so that they aren't loaded on every page. Fixed conflicts - I think it should be ready to go

geezhawk commented:
Whoops closed by accident :)

geezhawk commented:
1. Good catch - I'll go ahead and get rid of those
2. That was a mistake on my part - I was fixing merge conflicts and accidentally removed that link. I'll go ahead and add it back in.
3. Cool! I haven't used Jekyll before so I didn't know that it was feasible to make a sub-layout.

I should be able to address all of these and push changes sometime tonight. Thanks for the feedback :)

geezhawk commented:
@gwpjp Requested changes made. Let me know if there are any other changes you would like me to make!

gwpjp commented:
The new changes look great!  Looks like you forgot to do one item on the list.

1. There are a couple of holdovers from the original branch. Both css/main.css and images/email.png are superfluous files that have since been deleted. These can be removed.

Can you make this fix?

geezhawk commented:
My bad man - I thought you were referring to the extra files in the starters directory. I'll get rid of those other files when I'm free.

gwpjp commented:
It's bad form but I went ahead and merged it since I need to do some work on the website anyway.  I'll make the changes in the next commit.

geezhawk commented:
NP man thanks!
